hbase-client/src/main/java/org/apache/hadoop/hbase/client/CoprocessorDescriptorBuilder.java

```diff
@@ -111,5 +114,34 @@ public String toString() {
       return "class:" + className + ", jarPath:" + jarPath + ", priority:" + priority
         + ", properties:" + properties;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (this == obj) {
+        return true;
+      }
+      if (!(obj instanceof CoprocessorDescriptor)) {
+        return false;
+      }
+      CoprocessorDescriptor other = (CoprocessorDescriptor) obj;
+      if (
+        priority != other.getPriority() || !Objects.equals(className, other.getClassName())
+          || !Objects.equals(getJarPath(), other.getJarPath())
+      ) {
+        return false;
+      }
+      return Objects.equals(getProperties(), other.getProperties());
+    }
+
+    @Override
+    public int hashCode() {
+      int result = Objects.hash(className, jarPath, priority);
+      List<Map.Entry<String, String>> entries = new ArrayList<>(properties.entrySet());
```

Comment:
Ah, checked the code, properties is already a TreeMap... So we do not need to sort here, just iterate over it is enough...

To make it safe, we can change the declaration of properties from Map to SortedMap or NavigableMap.

Reply:
Thanks @Apache9 . I think after we move to NavigableMap we can change the whole hashCode() method to a simple Objects.hash(className, jarPath, priority, properties);

This will be exactly equivalent to current logic.

Reply:
Updated the PR. Have updated the Map to NavigableMap in interface as well. Let me know what you think about the same.